src/queue.c

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2018 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "queue_internal.h"
+#include "queue.h"
+#include "rebar-xxd.h"
+/*
+ queue.h implements a singly linked list FIFO
+ */
+
+/*
+ * Allocates memory for the queue, queue must be deleted by
+ * calling rebar_queue_delete(), all client data must have been 
+ * freed by the user of this API, Queue will only free it's own structures.
+ * Will assert in os if malloc() fails!
+ */
+queue_t *rebar_queue_init (void)
+{
+    queue_t *q = (queue_t *) malloc(sizeof(queue_t));
+    memset(q, 0, sizeof(queue_t));
+
+    return q;
+}
+
+/*
+ */
+int rebar_queue_delete (queue_t *q)
```

Comment:
You need a deleter function that accepts the user data as an argument so that it can delete the user data & not leak it.